x86: fix newlib with MMU enabled

tyrantlizard77 opened the pull request:
newlib uses any RAM between _end and the bounds of physical
RAM for the _sbrk() heap. Set up a user-writable boot region
so that this works properly.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

agross-linaro commented on include/arch/x86/linker.ld:
```diff
@@ -307,15 +307,15 @@ SECTIONS
-	_end = .; /* end of image */
+	MMU_PAGE_ALIGN
+	_end = .; /* end of image, used to begin the newlib heap */
+	_newlib_max_heap = (PHYS_RAM_ADDR + KB(CONFIG_RAM_SIZE)) - _end;
```
is this the size of the heap?  If so, the name needs to denote that.

andrewboie replied:
it's the maximum size of the heap. by default, the internal pointers tracked by _sbrk() point to _end, _newlib_max_heap is how far out it can ultimately go.

agross-oss reviewed (approved):
aside from the choice of name for a the max heap size, this looks fine.

Commit:
x86: grant user mode access to newlib heap

Newlib uses any RAM between _end and the bounds of physical
RAM for the _sbrk() heap. Set up a user-writable region
so that this works properly on x86.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
